### ISSUE TITLE
Context does not exist error during OIDC Federated Logout

### ISSUE THREAD

**Issue by chamathns** [wso2/product-is#9577]
**Describe the issue:**
The following error occurs while trying out the OIDC federated logout flow.
`
ERROR {org.wso2.carbon.identity.application.authentication.framework.handler.request.impl.DefaultRequestCoordinator} - Context does not exist. Probably due to invalidated cache. Requested client: 127.0.0.1, URI :GET:/commonauth, User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:80.0) Gecko/20100101 Firefox/80.0 , Referer: https://localhost:9443/
WARN {org.wso2.carbon.identity.application.authentication.framework.handler.request.impl.DefaultRequestCoordinator} - Authentication context is null, redirect parameter filtering will not be done for null
`

**How to reproduce:**
Setup OIDC federated authentication with two Identity Server Instances. [1][2]
Register a service provider(pickup-dispatch) and enable federated authentication.
Initiate the federated authentication flow by logging in via client application.
After logging in upon successful authentication, click on logout.

#### Trying the authentication flow with federated authentication
Upon clicking on logout via the client application, the federated logout request; `https://localhost:9444/oidc/logout`, will have the id_token_hint, state and post_logout_redirect_uri as the parameters.
Then, the application registered in the second server (idp running on port 9444) will redirect to its registered callback URL (https://localhost:9443/commonauth).
Here, the state parameter sent in the federated logout request does not get appended to the callback URL redirect of the idp, causing the error.
However, if we append the state parameter in the federated logout request to the callback URL redirect and manually send the request in the browser, (https://localhost:9443/commonauth?state=xxxxx) the logout flow would complete successfully without any errors.

#### Trying the authentication flow with default authentication
The issue with the federated logout scenario was, state parameter not being added to the callback URL in the redirect. In that case, the SP callback URL in the IDP did not get appended with the state parameter sent in the logout request. Hence, the authentication flow was carried out without federated authentication to test whether this issue persists in the default authentication flow as well. There, if we initiate a logout request manually with a state parameter added to the logout request, the state parameter was appended to the callback URL in the redirect, and the logout completed successfully.

**Environment information :**
 - Product Version: IS 5.10.0/5.11.0-alpha2

[1] - https://is.docs.wso2.com/en/latest/learn/configuring-oauth2-openid-connect/
[2] - https://medium.com/@methmas/federated-authentication-with-wso2-identity-server-with-openid-protocol-and-wso2-playground2-app-212151d66f7a

**Comment by emswbandara** [comment]
@Yasasr1 will look into this.

**Comment by ThaminduR** [comment]
This issue is due to the same host name in both Identity Servers. As cookies are set for the domain, both localhost:9443 and localhost:9444 will have the same commonAuthId cookie. During the logout 9443 will remove the commonAuthId cookie and 9444 won't have the commonAuthId cookie. 
We can avoid this error by using a different host name for federated identity server.

**Comment by ChamathNS** [comment]
The issue persists in scenarios where a federated idp is configured in different tenant of the same server. There, the different tenants will share the same domain and hence, will have the same commonAuthId cookie.

**Comment by kalyan312** [comment]
I using different server but facing the issue. Is there any other solution for the problem?

**Comment by sadilchamishka** [comment]
> The issue persists in scenarios where a federated idp is configured in different tenant of the same server. There, the different tenants will share the same domain and hence, will have the same commonAuthId cookie.

Can we reverse proxy the requests send for that tenant which holds the federated IDP and treat that as a separate domain.

**Comment by DilshanSenarath** [comment]
Tried to reproduce the error in [WSO2 Identity Server 6.2.0 Alpha3](https://github.com/wso2/product-is/releases/tag/v6.2.0-alpha3). But the OIDC Federated logout flow was successfully completed. Refer the following screen recording.

https://github.com/wso2/product-is/assets/74205483/017ac8a0-747b-44b2-a148-f21f9a33ddd4

I have checked this issue with IS Alpha-4 (Upcoming Release - Current changes) as well. In that case also, the OIDC Federated logout flow was successfully completed.
